stdlib/Distributed/src/remotecall.jl

```diff
@@ -351,7 +359,12 @@ end
 
 channel_type(rr::RemoteChannel{T}) where {T} = T
 
-serialize(s::ClusterSerializer, f::Future) = serialize(s, f, f.v === nothing)
+function serialize(s::ClusterSerializer, f::Future)
+    @lock f.lock begin
+        v_cache = @atomic f.v
+        serialize(s, f, v_cache === nothing)
+    end
```

Comment:
nobody can set this without holding the lock, right?
```suggestion
    @lock f.lock serialize(s, f, f.v === nothing)
```

Reply:
yes, setting is only done within the lock